packages/proxy/lib/http/response-middleware.ts

```diff
@@ -774,7 +774,26 @@ const MaybeRemoveSecurity: ResponseMiddleware = function () {
   this.next()
 }
 
-const GzipBody: ResponseMiddleware = function () {
+const GzipBody: ResponseMiddleware = async function () {
+  if (this.protocolManager && this.req.browserPreRequest?.requestId) {
+    const span = telemetry.startSpan({ name: 'gzip:body:protocol-notification', parentSpan: this.resMiddlewareSpan, isVerbose })
+    const resultingStream = this.protocolManager.responseStreamReceived({
+      requestId: this.req.browserPreRequest.requestId,
+      responseHeaders: this.incomingRes.headers,
+      isAlreadyGunzipped: this.isGunzipped,
+      responseStream: this.incomingResStream,
+      res: this.res,
+    })
+
+    if (resultingStream) {
+      this.incomingResStream = resultingStream.on('error', this.onError).once('finish', () => {
```

Comment:
I don't see the `finish` event on [Readable](https://nodejs.org/docs/latest-v16.x/api/stream.html#class-streamreadable)? Is this supposed to be `close`? Does finish/close still get called when there is an error? If not, we'll want to make sure the span also gets closed in that scenario.

Reply:
I was mirroring what was done a few lines later. Finish is on writable.

Reply:
Good call though I do wonder if this should be close

Reply:
Yes close gets called in both cases.